base/set.jl

```diff
@@ -17,6 +17,8 @@ Set() = Set{Any}()
 Construct a [`Set`](@ref) of the values generated by the given iterable object, or an
 empty set. Should be used instead of [`BitSet`](@ref) for sparse integer sets, or
 for sets of arbitrary objects.
+!!! note
+    Iteration over `Set` is unordered.
```

Comment:
I would say that the iteration is ordered, but that the order is unspecified.

But seems like the bigger issue here is that we don't define what a "Set" is. I think the Dict docstring is a better model for the text here.

```
    Set([itr]) <: AbstractSet
 
Construct an unordered collection of unique values as a `Set`. Values are compared with [`isequal`](@ref) and hashed with [`hash`](@ref).

Given a single iterable argument, constructs a `Set` whose values are generated by the object.

This type should be used instead of [`BitSet`](@ref) for sparse integer sets, or
for sets of arbitrary objects.
```